cypress/plugins/testDataProcessor.md

```diff
@@ -0,0 +1,233 @@
+ 
+ ## Functions
+ 
+### > testDataProcessor
+The Test Data Processor performs the following operations:
+- Merges all JSON files located within a specified directory.
+- Resolves the value for a specified key in each JSON object.
+
+Ex:
+
+```
+testDataProcessor()
+```
+
+### > processFireboltJson
+Iterating over the provided JSON and each object, resolve the values of params, context, or content if present, then returning the JSON with the updated value.
+
+##### Params:
+| Param | Definition|
+| --- | --- |
+| jsonData | json data for processing |
+
+Ex:
+
+```
+Request
+- processFireboltJson({'abc': {'method': 'method_name', 'params': 'TRUE'}})
+
+- processFireboltJson({'abc': {'method': 'method_name', 'params': 'TRUE', 'context': 'no_context'}})
+
+- processFireboltJson({'abc': {'method': 'method_name', 'validationJsonPath': 'result', 'content': 'TRUE'}})
+
+Response
+- {'abc': {'method': 'method_name', 'params': true}}
+
+- {'abc': {'method': 'method_name', 'params': true, 'context': {}}}
+
+- {'abc': {'method': 'method_name', 'validationJsonPath': 'result', 'content': true}}
+```
+
+### > testDataHandler
+This command is used mainly for parsing and fetching test data from various fixture files depending on the requestType passed to the command. RequestType can have different values based on what type of data need to be parsed and fetched from test data file. DataIdentifier specifies the key of the test data which is used to identify and fetch the test data from different test data files.
+
+Currently supported requestType is params, context and content.
+
+- Params - Represent the parameters to be sent for a firebolt call.
+
+  For the Params request type, the dataIdentifier can have two types.
+   * Key - Returns the value associated with the key `param_key` from the test data.
+   * Environment Variable Key - Returns the key as is, when it's having `CYPRESSENV`.
+
+- Context - Represent the data that needs to be stored in api/event object that helps to search specific object in a global list.
+
+- Content - Represent the expected value used for validating against a firebolt api response.
+
+  Currently two types of validation are there
+   * Error validation
+   * Result validation
+    - Static Content Validation - Required content will be fetched from combined(fcs and config module) `default json` or from the corresponding `module json` files based on the precedence.
+      - `module.json` files reside in `fixtures/modules/` or `nodemodules/configModule/testData/modules/`
+
+    - Device Content Validation
+      - Content will be fetched from `devicMac.json` file when devicMac is present. 
+      - `deviceMac.json` present in `nodemodules/configModule/testData/devices/`.
+      - deviceMac is not present content will be taken from `defaultDeviceData` json file.
+
+Note:
+ - When data is not present in the fixtures, returning the passed the key as is.
+ - `Mock Firebolt` is the default device and mac addressed is not required.
+   
```

Comment:
Mock Firebolt is the default device and mac addressed is not required.
  
Not sure i understood above. Can you elaborate or make the sentence better

Reply:
Removed this line, because for mfos also mac id is required